### DISCUSSION TITLE
Inverter - Total Energy

### DISCUSSION THREAD

**Discussion by avannigt** [addiejanssen/domoticz-solaredge-modbustcp-plugin#11] [Q&A]
Hi, I am new to Github and have a question. I installed domoticz-solaredge-modbustcp-plugin. That all went fine. I then installed the hardware in Domoticz, after which various devices became visible in the devices page. So far good. However, I found that the number of KW in Inverter Total Enery in the device screen does not match the number of KW listed on the Other tab. Is there any way to fix this?

**Comment by addiejanssen** [comment]
Hi @avannigt,

Can you share some screenshots of the issue?

Thanks,

Addie

**Reply by avannigt** [reply]
![1](https://user-images.githubusercontent.com/81971401/113588404-6579a680-9630-11eb-8dda-ae096ebbfc9e.jpg)
![2](https://user-images.githubusercontent.com/81971401/113588423-6ad6f100-9630-11eb-9fcc-657596e53649.jpg)

**Comment by addiejanssen** [comment]
Thanks for the screenshots.

The short answer is: this is correct 😊 There is nothing to fix.

The long answer:

The number on the device page shows the total energy that the inverter has generated over time.
So, basically from day one when it was turned on. In your case, 7818.173 kWh.
Not sure if you are able to login to the [SolarEdge monitoring portal](https://monitoring.solaredge.com/), but if you would, you should see that number as “total energy” over there as well. Be aware, the information on the SolarEdge monitoring portal is updated at other intervals as the Domoticz plugin and as a result, the numbers may not always be a perfect match. If you want to compare, then do so at night when the inverter is sleeping.

The total energy as reported on the utility page (13.474 kWh) is calculated using that total energy figure from the inverter.
When you subtract the value as shown on the device page at a particular moment during the day and the value at the start of the day, you should get the number as shown on the utility page. This is what Domoticz is doing for you.

The bold figure on the utility page (2564.39 Watt) will constanly change. It shows the power being generated at the moment the sample is taken from the inverter.

Hope this explains what you see; no worries: this is how it should be.

Kind regards,

Adddie

**Reply by avannigt** [reply]
Thanks for your quick reply. But I am left with the question whether it is not possible to display the total energy generated by the inverter. Ultimately, this value (7518 KW) is in the device screen of Domoticz and should therefore also be visible on the utility page. Or I don't understand.

**Reply by addiejanssen** [reply]
OK, got your point.
At this moment, there is no meter configured by the plugin to do that.

Can it be done? Yes sure. However, I wonder how/what to show.
Most of the values and graphs in Domoticz are relative to time. They show what was generated today or last year.
But this number only goes up and over time, the graph will become boring: it would look like a single line going up....
And since the number will get bigger and bigger, due to scaling of the graph, you would only see a flat line over time.

**Comment by avannigt** [comment]
That is indeed correct. But for me it is not about the graph but to be able to see the total generated energy at a glance on the utility page. If you know what I mean.

**Reply by jvanderzande** [reply]
This is pretty easy to accomplish by combining the I_AC_Power and the I_AC_Energy_WH values into one request to Domoticz, where you update the device with this JSON request:
https://www.domoticz.com/wiki/Domoticz_API/JSON_URL%27s#Electricity_.28instant_and_counter.29
I have combined them in this way in NODE-RED:
```
Watts = Number(msg.payload.I_AC_Power) * Math.pow(10, msg.payload.I_AC_Power_SF);
WattH = Number(msg.payload.I_AC_Energy_WH) * Math.pow(10, msg.payload.I_AC_Energy_WH_SF);
Prev_SolarEdge_WattH = Number(flow.get("Prev_WattH"));
// stop in case the new WattH is lowe than the previous value as this never should happen.
if (Prev_SolarEdge_WattH > WattH) {
    node.error("Current WattH (" + WattH + ") lower than Prev_WattH (" + Prev_WattH + ")");
    return [null,null]
} 
flow.set("Prev_SolarEdge_WattH", WattH);
msgo = {};
msgo.payload={};
msgo.method="get";
msgo1.url='http://192.168.0.31:8080/json.htm?type=command&param=udevice&idx=509&svalue=' + Watts + ';' + WattH;

```

**Reply by avannigt** [reply]
That is too difficult for me. I am a novice. I do see the correct number of "Total kWh" in the devices page. Couldn't this be made easier to view on the Domoticz "Other" screen?

**Reply by jvanderzande** [reply]
Sorry, I understand now and was confused as I though you were using Node-red, but this is obviously about the plugin.   
So simply ignore my previous comments. ;)

**Reply by avannigt** [reply]
As i wrote;
The total energy is in the "apparaten" screen. See picture 1. But the amount is wrong in the "overige" screen. See picture 2.
How do i fix that ?

![Devices](https://user-images.githubusercontent.com/81971401/116718606-6e188f00-a9da-11eb-87b3-d3764d6ebb52.jpg)
![Overige](https://user-images.githubusercontent.com/81971401/116718612-707ae900-a9da-11eb-8e36-8ebb15fd2932.jpg)

**Reply by jvanderzande** [reply]
I understood but am still in the dark how that data is send from NODERED to Domoticz. What logic are you using in the flow ....or how does your flow look?

**Reply by avannigt** [reply]
I am not a programmer and unfamiliar with writing scripts and flows. I have connected the inverter via SolarEdge Lan and SolarEdge Modbus in Domoticz. I just want the Total Energy value to be properly displayed in my dashboard and not just the devices screen.
